assertj-core/src/test/java/org/assertj/core/api/bigdecimal/BigDecimalsAssert_assertHasPrecisionOf_Test.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2023 the original author or authors.
+ */
+package org.assertj.core.api.bigdecimal;
+
+import org.assertj.core.api.BigDecimalAssert;
+import org.assertj.core.api.BigDecimalAssertBaseTest;
+
+import static org.mockito.Mockito.verify;
+
+class BigDecimalsAssert_assertHasPrecisionOf_Test extends BigDecimalAssertBaseTest {
+
+  @Override
+  protected BigDecimalAssert invoke_api_method() {
+    return assertions.hasPrecisionOf(1);
+  }
+
+  @Override
+  protected void verify_internal_effects() {
+    verify(bigDecimals).assertHasPrecision(getInfo(assertions), getActual(assertions), 1);
+  }
+
+}
```

Comment:
Once the implementation is pulled up, this test class should contain actual test cases like `CharSequenceAssert_isVisible_Test` (and there is no need to extend `BigDecimalAssertBaseTest`).